src/main/java/com/github/mtakaki/credentialstorage/hibernate/UnitOfWorkAspect.java

```diff
@@ -0,0 +1,138 @@
+package com.github.mtakaki.credentialstorage.hibernate;
+
+import java.util.Map;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.context.internal.ManagedSessionContext;
+
+import lombok.RequiredArgsConstructor;
+
+/**
+ * An aspect providing operations around a method with the {@link UnitOfWork}
+ * annotation. It opens a Hibernate session and optionally a transaction.
+ * <p>
+ * It should be created for every invocation of the method.
+ * </p>
+ */
+@RequiredArgsConstructor
+public class UnitOfWorkAspect {
+    private final Map<String, RemoteCredentialHibernateBundle<?>> bundles;
+
+    // Context variables
+    private UnitOfWork unitOfWork;
+    private Session session;
+    private SessionFactory sessionFactory;
+    private RemoteCredentialHibernateBundle<?> bundle;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'bundle' could be replaced by a local variable.](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455857651/issues/source?bid=3237716&fileBranchId=3250434#l27)